src/Microsoft.Extensions.Options/IConfigureNamedOptions.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.Extensions.Options
+{
+
+    /// <summary>
+    /// Represents something that configures the TOptions type.
+    /// </summary>
+    /// <typeparam name="TOptions"></typeparam>
+    public interface IConfigureNamedOptions<in TOptions> where TOptions : class
```

Comment:
So old APIs that exist should behave the same.  We are striving for complete backwards compatibility for any apps that don't use any of the new APIs.  That said we are free to modify the behavior of any old or new APIs to light up with the new IOptionsFactory.Get, and we can also where it makes sense have any new ConfigureAll APIs also register the old configures so they show up.  Its likely we will need to think carefully about how all of these interact, which is why to start with they are each on their own islands basically.

If you want to configure two named options the same way, you can just a helper method and call that in your configure `services.Configure("name", o => DoStuff())`

For your tweak the default scenario, that's why ConfigureAll is there.  call ConfigureAll to setup everything, then call Configure("name") to tweak